account/payments.go

```diff
@@ -340,69 +618,106 @@ func (a *Service) getMaxAmount(destination string, routeHints []*lnrpc.RouteHint
 		a.log.Errorf("lnclient.ListChannels error: %v", err)
 		return 0, fmt.Errorf("lnclient.ListChannels error: %w", err)
 	}
+
 	var totalMax uint64
+OUTER:
 	for _, c := range channels.Channels {
-		if c.LocalBalance == 0 {
+		// Calculate how much can be sent through this channel after accounting for channel reserve
+		thisChannelCanSendSat, minHtlcMsat := channelConstraints(c)
+
+		if thisChannelCanSendSat*1000 <= int64(minHtlcMsat) {
 			continue
 		}
-		a.log.Infof("cid: %v, active: %v, balance: %v, channnel reserve: %v", c.ChanId, c.Active, c.LocalBalance, c.LocalConstraints.ChanReserveSat)
-		_, err := lnclient.QueryRoutes(context.Background(), &lnrpc.QueryRoutesRequest{
+
+		a.log.Infof("cid: %v, active: %v, balance: %v, can send: %v, channel reserve: %v, minHtlcMsat: %v", c.ChanId, c.Active, c.LocalBalance, thisChannelCanSendSat, c.LocalConstraints.ChanReserveSat, minHtlcMsat)
+
+		// Try to find a route for the maximum amount we can send through this channel
+		// Start with a small probe amount to check if the route exists
+		routes, err := lnclient.QueryRoutes(context.Background(), &lnrpc.QueryRoutesRequest{
 			PubKey:         destination,
-			Amt:            c.LocalBalance - int64(c.LocalConstraints.ChanReserveSat) + 1,
+			AmtMsat:        int64(minHtlcMsat),
```

Comment:
Curious why we are not using the channel balance here as a first step like we do in sendPayment?

Reply:
Note that sendPayment doesn't do that anymore.
We used to have a modification in the pathfinding algorithm in LND that returned a specific error when no route could be found. It said 'try lowering the amount to x'. We don't have that inside LND anymore, so querying a route with the channel balance will return no route. This is because the amount passed here is the destination amount. If the channel balance is passed then it will always return 'no route' if there's fees along the path.

Reply:
I see now yes, the send payment only attempts to query route for the available amount and if failed it fallbacks to query route with minimal amount and then calculate the fee manually. 
Question: What is the advantage of the first attempt? Why not always try the manual approach from the first place?

Reply:
With first attempt you mean the part at the top where we do a queryRoute with the destination amount rather than constructing the path ourselves?
This is because for the 'last' chunk, you want to start with the destination amount and work your way back from there. So that's basically regular pathfinding. It's not necessarily the maximum amount we're trying to send in this function. It's the amount passed in by the caller.